lib/SILOptimizer/Mandatory/DefiniteInitialization.cpp

```diff
@@ -1047,7 +1047,15 @@ void LifetimeChecker::handleStoreUse(unsigned UseID) {
   if (isFullyUninitialized) {
     Use.Kind = DIUseKind::Initialization;
   } else if (isFullyInitialized) {
-    Use.Kind = DIUseKind::Assign;
+    // Only re-write assign_by_wrapper to assignment if all fields have been
+    // initialized.
+    if (isa<AssignByWrapperInst>(Use.Inst) &&
+        getAnyUninitializedMemberAtInst(Use.Inst, 0,
```

Comment:
Do you still need the isFullyUninitialized check above? Unless I'm missing something, it seems like "property wrapper is uninitialized" would be a special case of "any field uninitialized".

Reply:
That case is still needed for non-property-wrapped properties

Reply:
Sounds like we need to use assign_by_wrapper for everything then, but maybe not in this PR :-)